rust/lance-core/src/encodings/plain.rs

```diff
@@ -396,6 +396,10 @@ fn make_chunked_requests(
     //    indices[i] * byte_width + block_size < indices[i+1] * byte_width
     // It might allow slightly larger sequential reads.
     for i in 0..indices.len() - 1 {
+        // If contiguous, continue
+        if indices[i + 1] == indices[i] {
+            continue;
+        }
```

Comment:
This fixes a performance bug for large vectors. If we have vectors larger than the block size (1536-dim f32 vectors are 6,144 bytes while the default block size on local fs is 4K), then without this line we make a range request for **every** vector we take. 😱